packages/cyberstorm/src/components/CustomDropDown.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ReactNode } from "react";
+import styles from "./componentStyles/CustomDropDown.module.css";
+import { Button } from "./Button";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+import { SelectItemProps } from "./CustomSelectItem";
+
+type DropDownProps = {
+  colorScheme?: "default" | "defaultDark" | "primary";
+  defaultOpen?: boolean;
+  icon?: ReactNode;
+  content?: ReactNode[];
+  label?: string;
+};
+
+export const CustomDropDown: React.FC<DropDownProps> = (props) => {
+  const { colorScheme, defaultOpen, icon, content, label } = props;
+
+  return (
+    <div className={styles.root}>
+      <DropdownMenu.Root defaultOpen={defaultOpen}>
+        <DropdownMenu.Trigger asChild>
+          <div>
```

Comment:
[Ref comment](https://github.com/thunderstore-io/thunderstore-ui/pull/599#discussion_r1066863646)

> `<div>` can't be omitted, Radix's asChild doesn't like custom components, so those have to be wrapped.

[This issue on Radix GitHub](https://github.com/radix-ui/primitives/issues/953) tells that the components used with `asChild` need to spread DOM props to work. Is this something we should and can do?

This applies to all components built with Radix, including `CustomSelect` in this PR.

Reply:
Got it this time, should be OK now.

Reply:
I'm resolving this since this has been worked on and there's another, more up-to-date discussion involving this in another comment.